terraform/modules/iotedgevm/main.tf

```diff
@@ -0,0 +1,99 @@
+resource "random_string" "vm_user_name" {
+  length  = 10
+  special = false
+}
+
+resource "random_password" "vm_user_password" {
+  length           = 10
+  min_upper        = 1
+  min_lower        = 1
+  min_special      = 1
+  min_numeric      = 1
+  override_special = "!@#$%&"
+}
+
+locals {
+  dns_label_prefix = "${var.name_identifier}-iot-edge"
+  vm_username      = var.vm_user_name != "" ? var.vm_user_name : random_string.vm_user_name.result
+  vm_userpassword  = var.vm_user_password != "" ? var.vm_user_password : random_password.vm_user_password.result
+}
+
+resource "azurerm_public_ip" "iot_edge" {
+  name                = "${local.dns_label_prefix}-ip"
+  resource_group_name = var.rg_name
+  location            = var.location
+  allocation_method   = "Dynamic"
+  domain_name_label   = "${local.dns_label_prefix}-${var.random_id}"
+}
+
+resource "azurerm_network_security_group" "iot_edge" {
+  name                = "${local.dns_label_prefix}-nsg"
+  resource_group_name = var.rg_name
+  location            = var.location
+
+  security_rule {
```

Comment:
Should we leave SSH port open by default?

Reply:
Good question. But I don't know. If we delete it, then a user won't be able to access the VM with the current setup. But not sure if that's really needed in the ELMS. Probably not. And it's more secure without it.

Reply:
We decided to not leave SSH port open. If needed user can find documentation on how to access a VM.